paparazzi/src/main/java/app/cash/paparazzi/PaparazziSdk.kt

```diff
@@ -285,21 +294,41 @@ public class PaparazziSdk @JvmOverloads constructor(
       viewGroup.addView(modifiedView)
       for (frame in 0 until frameCount) {
         val nowNanos = (startNanos + (frame * 1_000_000_000.0 / fps)).toLong()
-        withTime(nowNanos) {
-          val result = renderSession.render(true)
-          if (result.status == ERROR_UNKNOWN) {
-            throw result.exception
-          }
 
-          val image = bridgeRenderSession.image
-          if (validateAccessibility) {
-            require(renderExtensions.isEmpty()) {
-              "Running accessibility validation and render extensions simultaneously is not supported."
+        // If we have pendingTasks run recomposer to ensure we get the correct frame.
+        var result = withTime(nowNanos) {
```

Comment:
```suggestion
        var hasPendingWork = false
        withTime(nowNanos) {
          val result = renderSession.render(true)
          if (result.status == ERROR_UNKNOWN) {
            throw result.exception
          }

          if (hasComposeRuntime && recomposer != null) {
            // If we have pending tasks, we need to trigger it within the context of the first frame.
            if (frame == 0 && (recomposer as Recomposer).hasPendingWork) {
              hasPendingWork = true
            }
          }
        }
```

then within the same loop iteration of `frame`:

```
        if (hasPendingWork) {
          withTime(nowNanos) {
            val result = renderSession.render(true)
            if (result.status == ERROR_UNKNOWN) {
              throw result.exception
            }
          }

          val recomposerInstance = recomposer as Recomposer
          if (recomposerInstance.hasPendingWork) {
            logger.warning(
              "Pending work detected. This may cause unexpected results in your generated snapshots. ${recomposerInstance.changeCount}"
            )
          }
        }
```